### PR TITLE
tests for #457, #458

### PR DESCRIPTION
**Pull request by TheoPannetier**
**Tests for:**
- #457 
- #458

### CODE REVIEW

File: game.cpp [hunk 1]
```diff
@@ -1902,6 +1902,33 @@ void test_game() //!OCLINT tests may be many
...
+
+//#define FIX_ISSUE_458
+#ifdef FIX_ISSUE_458
+  {
+    // (458) The color of any food item can be accessed easily
+    game g;
+    color color_food = get_nth_food_color(g, 0);
+    // Default color for the food item is black
+    color c;
+    assert(color_food == c);
```
Comment (richelbilderbeek) [inline game.cpp:1928]: This is scary, as the comment is needed to explain the code, where these may very likely change in the future.

```c++
assert(color_food == color::black);
assert(color_food == create_black());
assert(color_food == create_black_color());
```

File: game.cpp [hunk 2]
```diff
@@ -1902,6 +1902,33 @@ void test_game() //!OCLINT tests may be many
...
+    assert(color_player_two == create_green_color());
+    assert(color_player_three == create_blue_color());
+  }
+#endif
+
+//#define FIX_ISSUE_458
+#ifdef FIX_ISSUE_458
+  {
+    // (458) The color of any food item can be accessed easily
+    game g;
```
Comment (richelbilderbeek) [inline game.cpp:1924]: As we only read from the `game`, it can (and hence should) be `const game g;`

File: game.cpp [hunk 3]
```diff
@@ -1902,6 +1902,33 @@ void test_game() //!OCLINT tests may be many
   }
   #endif // FIX_ISSUE_241
 
+//#define FIX_ISSUE_457
+#ifdef FIX_ISSUE_457
+  {
+    // (457) The color of any player can be accessed easily
+    game g;
```
Comment (richelbilderbeek) [inline game.cpp:1909]: As we only read from the `game`, it can (and hence should) be `const game g;`

File: game.cpp [hunk 4]
```diff
@@ -1902,6 +1902,33 @@ void test_game() //!OCLINT tests may be many
...
+    color color_player_one = get_nth_player_color(g, 0);
+    color color_player_two = get_nth_player_color(g, 1);
```
Comment (richelbilderbeek) [inline game.cpp:1911]: All these colors can (and hence should) be `const`

**Review by richelbilderbeek — CHANGES_REQUESTED** [review]
Hi @TheoPannetier,

Great work! 

I have been a bit more strict as I think you will like that and hence request to be more const-correct, especially [Con.1: By default, make objects immutable](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rconst-immutable).

So, get those `const`s in and I'd be happy to accept!

File: game.cpp (hunk 1 above)
Comment (TheoPannetier) [reply game.cpp]: The test now expects the food color to be the default color (of the class). No need for a possibly lying comment anymore!

**Comment by TheoPannetier** [comment]
I've sneaked a new test in, for #460

**Review by richelbilderbeek — APPROVED** [review]
Increased const-correctness, I like that! Note that `is_first_color_winner` could also be const-corrected, now it is time to get this on develop :-)

**Comment by richelbilderbeek** [comment]
Well done @TheoPannetier !

**Pull request merged**